sopel/modules/emailcheck.py

```diff
@@ -0,0 +1,314 @@
+    config.emailcheck.configure_setting('exempt_suffixes',
+                                        'Suffixes (TLD, whole domain, etc.) to exempt from checking')
+    config.emailcheck.configure_setting('warn_chans',
+                                        'List of channels to warn when a suspicious user is detected. '
+                                        'May be empty.')
+    config.emailcheck.configure_setting('protect_chans',
+                                        'List of channels to +R after malicious attempt to reg. '
+                                        'May be empty.')
+
+def setup(bot):
+    bot.config.define_section('emailcheck', EmailCheckSection)
+
+@dataclass
+class Email:
+    user: str
+    domain: str
+    def get_address(self):
+        return f'{self.user}@{self.domain}'
+    def __str__(self):
+        return self.get_address()
+    def __post_init__(self):
+        self.domain = self.domain.lower()
+
+@dataclass
+class DomainInfo:
+    score: float
+    flag_disposable: bool
+    flag_recent_abuse: bool
+
+def alert(bot, alert_msg: str, log_err: bool = False):
+    for channel in bot.config.emailcheck.warn_chans:
+        bot.say(alert_msg, channel)
+    if log_err:
+        LOGGER.error(alert_msg)
+
+def add_badmail(bot, email):
+    #Right now we're BADMAILing whole domains. This might change.
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would badmail {email}")
+    else:
+        bot.write("NICKSERV", "badmail", "add", f'*@{email.domain}')
+
+def fdrop(bot, nick: str):
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would fdrop {nick}")
+    else:
+        bot.write("NICKSERV", "fdrop", nick.lower())
+
+def gline_ip(bot, ip: str, duration: str):
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would gline {ip} for {duration}")
+    else:
+        bot.write("GLINE", f'*@{ip}', duration, KILL_STR)
+
+def gline_username(bot, nick: str, duration: str):
+    if known_user := bot.users.get(Identifier(nick)):
+        username = known_user.user.lower() # Should already be lowercase
+        if IRCCLOUD_USER_REGEX.match(username):
+            if safe_mode:
+                LOGGER.info(f"SAFE MODE: Would gline {username} for {duration}")
+            else:
+                bot.write("GLINE", f'{username}@*', duration, KILL_STR)
+            return
+        else:
+            alert(bot, f"User {nick} had unexpected non-IRCCloud username {username}", true)
+    else:
+        alert(bot, f"Couldn't find irccloud uid/sid for {nick} to G-line!", true)
+    kill_nick(bot, nick) # Something went wrong with G-line, so fall back to /kill
+
+def kill_nick(bot, nick: str):
+    if safe_mode:
+        LOGGER.info(f"SAFE MODE: Would kill {nick}")
+    else:
+        bot.write("KILL", nick.lower(), KILL_STR)
+
+def gline_strategy(bot, nick):
+    if (known_user := bot.users.get(Identifier(nick))):
+        if hasattr(known_user, "ip"):
+            ip = known_user.ip
+            exemption = get_exemption(ip)
```

Comment:
Oh I see now, `get_exemption()` is a function from your code, not a third-party library. In that case, I don't really understand what the `try-except` is there for.

Also (assuming you're using python 3) you should import local modules as `from . import local_module` or `from .local_module import some_function`

Reply:
It's so I could keep things running if I had to completely disable ip for whatever reason

So "from .ip import get_exemption"? Sounds good